File: Python/Day01.py
string = open('../inputs/Day01.txt').read()

floor = 0
basementPosition = 0

for move in enumerate(string, start = 1):
	floor += 1 if (move[1] == '(') else -1
	basementPosition = move[0] if floor == -1 and basementPosition == 0 else basementPosition

print "Part 1:", floor
print "Part 2:", basementPosition

File: JavaScript/Day01.js
(function(){
	"use strict";
	
	require("fs").readFile("../inputs/Day01.txt", "utf8", function(err, data){
		if(err){
			console.log(`Unable to load file: ${err}`);
		} else {
			let basementPosition = 0;

			let floor = data.split('').reduce((s, v, i) => {
				if(s === -1 && basementPosition === 0){
					basementPosition = i;
				}
				return s + ((v == "(") ? 1 : -1);
			}, 0);

			console.log(`Part 1: ${floor}`);
			console.log(`Part 2: ${basementPosition}`);
		}
	});
})();